Workspace Editor Page Design Updates

dangrous opened the pull request:
<!-- If necessary, assign reviewers that know the area or changes well. Feel free to tag any additional reviewers you see fit. -->

### Details
<!-- Explanation of the change or anything fishy that is going on -->

### Fixed Issues
<!---
1. Please replace GH_LINK with a URL link to the GitHub issue this Pull Request is fixing.
2. Please replace PROPOSAL: GH_LINK_ISSUE(COMMENT) with a URL link to your GitHub comment, which contains the approved proposal (i.e. the proposal that was approved by Expensify).

Do NOT add the special GH keywords like `fixed` etc, we have our own process of managing the flow.
It MUST be an entire link to the github issue and your comment proposal ; otherwise, the linking will not work as expected.

Make sure this section looks similar to this (you can link multiple issues using the same formatting, just add a new line):

$ https://github.com/Expensify/App/issues/<number-of-the-issue>
$ https://github.com/Expensify/App/issues/<number-of-the-issue(comment)>

Do NOT only link the issue number like this: $ #<number-of-the-issue>
--->
$ https://github.com/Expensify/App/issues/12257 

### Tests
<!---
Add a numbered list of manual tests you performed that validates your changes work on all platforms, and that there are no regressions present.
Add any additional test steps if test steps are unique to a particular platform.
Manual test steps should be written so that your reviewer can repeat and verify one or more expected outcomes in the development environment.

For example:
1. Click on the text input to bring it into focus
2. Upload an image via copy paste
3. Verify a modal appears displaying a preview of that image
--->
* See QA Steps

- [x] Verify that no errors appear in the JS console

### QA Steps
<!---
Add a numbered list of manual tests that can be performed by our QA engineers on the staging environment to validate that your changes work on all platforms, and that there are no regressions present.
Add any additional QA steps if test steps are unique to a particular platform.
Manual test steps should be written so that the QA engineer can repeat and verify one or more expected outcomes in the staging environment.

For example:
1. Click on the text input to bring it into focus
2. Upload an image via copy paste
3. Verify a modal appears displaying a preview of that image
--->
Note: I'm sorry in advance. The testing steps are the same on all platforms. There should be no changes to functionality with any of these code changes. I'm a little unsure if these will actually get to all the screens - I had to hack a bit locally to make it work - so please reach out to me if you need assistance and we can work through it together.

* Log into an account without a VBA.
  * Create a new Workspace and go to the Workspace Settings
    * Click the question mark in the top right of the Workspace Settings view and confirm that the design matches 1 in the screenshots file, and all buttons/links work as expected.
      * Go to `Request a setup call` and confirm that the design matches 2 in the screenshots file, and all buttons/links work as expected.
    * Go to `Issue cards` and confirm that the design matches 10 in the screenshots file, and all buttons/links work as expected.
    * Go to `Reimburse expenses` and confirm that the design matches 16, 17, and 19 in the screenshots file, and all buttons/links work as expected.
    * Go to `Pay bills` and confirm that the design matches 7 and 8 in the screenshots file, and all buttons/links work as expected.
    * Go to `Send invoices` and confirm that the design matches 13 and 14 in the screenshots file, and all buttons/links work as expected.
    * Go to `Book travel` and confirm that the design matches 20 in the screenshots file, and all buttons/links work as expected.
    * Go to `Connect bank account` and confirm that the design matches 3 in the screenshots file, and all buttons/links work as expected.
    * Connect a bank account via Plaid and continue to the end of the flow, where it asks you to add a work email address. Confirm that the design matches 5 in the screenshots file, and all buttons/links work as expected
    * Cancel adding that bank account and restart the connection manually.
    * When you get to the flow asking for verification amounts, leave the flow.
    * Go back into Workspace settings and click `Connect bank account` again. Confirm that the design matches 4 in the screenshots file, and all buttons/links work as expected.
    * Return to the validation screen and enter in the amounts and continue. Confirm that the design matches 6 in the screenshots file, and all buttons/links work as expected.
* Log into an account/workspace with a VBA.
  * Go to the Workspace Settings
    * Go to `Issue cards` and confirm that the design matches 11 in the screenshots file, and all buttons/links work as expected.
    * Go to `Reimburse expenses` and confirm that the design matches 16, 17, and 18 in the screenshots file, and all buttons/links work as expected.
    * Go to `Pay bills` and confirm that the design matches 7 and 9 in the screenshots file, and all buttons/links work as expected.
    * Go to `Send invoices` and confirm that the design matches 13 and 15 in the screenshots file, and all buttons/links work as expected.
    * Go to `Book travel` and confirm that the design matches 21 in the screenshots file, and all buttons/links work as expected.
* Log into an account/workspace with a VBA and an Expensify Card
  * Go to the Workspace settings
  * Go to `Issue cards` and confirm that the design matches 12 in the screenshots file, and all buttons/links work as expected.
* Log into an account/workspace with an Expensify Card.
  * Go to the Workspace Settings
  * Go to `Connect a bank account` and go through the flow adding a bank account via Plaid.
  * After verifying your personal information, confirm that the design matches 22 in the screenshots file, and all buttons/links work as expected.

- [x] Verify that no errors appear in the JS console

### Offline Steps
* No changes are expected, offline elements should be greyed out; errors on relaunch should be displayed as before.

### PR Author Checklist
<!--
This is a checklist for PR authors. Please make sure to complete all tasks and check them off once you do, or else your PR will not be merged!
-->

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [X] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [X] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).

- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] iOS / native
    - [x] Android / native
    - [x] iOS / Safari
    - [x] Android / Chrome
    - [x] MacOS / Chrome
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [x] I verified any copy / text that was added to the app is correct English and approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

<details>
<summary><h4>PR Reviewer Checklist</h4>

The reviewer will copy/paste it into a new comment and complete it after the author checklist is completed
</summary>

- [ ] I have verified the author checklist is complete (all boxes are checked off).
- [ ] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] iOS / native
    - [ ] Android / native
    - [ ] iOS / Safari
    - [ ] Android / Chrome
    - [ ] MacOS / Chrome
    - [ ] MacOS / Desktop
- [ ] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [ ] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [ ] I verified any copy / text that was added to the app is correct English and approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [ ] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn't already exist
    - [ ] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

</details>

### Screenshots
<!-- Add screenshots for all platforms tested. Pull requests won't be merged unless the screenshots show the app was tested on all platforms.-->

All screenshots are available here: https://www.dropbox.com/s/1nv9cicwl758ea7/Screenshots.zip?dl=0

#### Web
<!-- Insert screenshots of your changes on the web platform-->

#### Mobile Web - Chrome
<!-- Insert screenshots of your changes on the web platform (from chrome mobile browser)-->

#### Mobile Web - Safari
<!-- Insert screenshots of your changes on the web platform (from safari mobile browser)-->

#### Desktop
<!-- Insert screenshots of your changes on the desktop platform-->

#### iOS
<!-- Insert screenshots of your changes on the iOS platform-->

#### Android
<!-- Insert screenshots of your changes on the Android platform-->

Commit:
initial updates for style on workspace editor page

shawnborton commented on src/styles/styles.js:
```diff
@@ -2761,6 +2761,20 @@ const styles = {
     },
+
+    workspaceSection: {
+        backgroundColor: colors.gray1,
```
I wonder if we can make a more global value here that comes from the theme file... something like `themeColors.cardBackground` or something? Then in the theme file, use `gray1`

grgia replied:
Yes, ideally we don't want to import colors anywhere except into the theme file. I'll be working on this when I update the colors in the upcoming week, but feel free to add as a variable to`default.js` for now!

dangrous replied:
Sounds good!

shawnborton commented on src/styles/styles.js:
```diff
@@ -2761,6 +2761,20 @@ const styles = {
         ...spacing.ph2,
     },
+
+    workspaceSection: {
```
I think this is fine for now, but I could see making this even more generic... like `cardSection` or something? In case we use it elsewhere

dangrous replied:
yeah I like that too. This was already more genericized than my first attempt (`workspaceEditorSection`) but will make it even broader.

shawnborton commented on src/styles/styles.js:
```diff
@@ -2761,6 +2761,20 @@ const styles = {
+    workspaceSection: {
+        backgroundColor: colors.gray1,
+        borderRadius: variables.componentBorderRadiusCard,
+        marginVertical: 10,
```
Maybe we can just do bottom margin for all of these cards? This way the first card in a view will sit more towards the top.

dangrous replied:
oh that works

grgia commented on src/pages/workspace/reimburse/WorkspaceReimburseView.js:
```diff
@@ -193,18 +194,21 @@ class WorkspaceReimburseView extends React.Component {
                                 icon: Expensicons.Receipt,
                                 shouldShowRightIcon: true,
                                 iconRight: Expensicons.NewWindow,
+                                iconFill: themeColors.buttonSuccessBG,
```
if we're holding for new color theme, you'll want to use `themeColors.success`

grgia replied:
You can wait on this until the colors are merged, I just wanted to make a note

dangrous replied:
Cool! Will keep this in here for now for testing purposes but can swap it once that's ready to go.

Commit:
update pages and consolidate images

dangrous commented:
Updated! I think I caught everything in the Figma you shared. There are a couple more places in settings where we use the same `Section` component, which I think are also the places we might want to update illustrations. They are:
* Enable Step (I think this is in the bank account setup flow but I'm not sure)
* Validation Step (I think this is in the bank account setup flow but I'm not sure)
* Workspace Bank Account Page
* Workspace Bills VBA View
* Workspace Card No VBA View
* Workspace Card VBA No ECard View
* Workspace Card VBA With ECard View
* Workspace Invoices VBA View
* Workspace Reimburse View (there's a fourth card for if they have a VBA)
* Workspace Travel VBA View

I think all of those will show up on an account that has a bank account connected - except for `Workspace Card No VBA View` which you may be able to get to now. I *think* the `Card` is referencing the Expensify card? I will try to get some screenshots of the existing design of the others so you can work on what you want instead!

There are also a couple places we use `Section`s that AREN'T settings related. I think it still would likely be worth changing these as well, but would love to know what you think. If we do update them, I'll go through this code and put all the styles into the component as defaults instead of passing it each time here. Those places are:
* Get Assistance Page
* Request Call Page

Also, I didn't update the button shapes (since that will be global, right?) or the icons (since I don't have the icons yet) but happy to do either/both if we have the goods.

And, let me know if you want me to rename any of the illustrations; these should mostly be fine but some are a little wonky (if we're eventually going to remove the existing ones we can be a little more broad in how we name these).

Lastly, there are some merge conflicts, I'll work on those separately.

shawnborton commented:
Yup, no worries on the button shape or icons just yet. 

For those sections you referenced, can you share screenshots of them here?

dangrous commented:
@shawnborton here you go! There are a couple I couldn't quite figure out how to get at yet, so I screenshotted the code (that notes what icons are used). I realize that's not super helpful but figured you could start on the others and then I will keep figuring out how to see the remaining four.

[Remaining Sections.zip](https://github.com/Expensify/App/files/9974623/Remaining.Sections.zip)

Also please remind me that in this PR i changed the background color of the sections to blue so I could see them better, so I need to update that before we go much further.

Commit:
prep to fix merge conflicts

Commit:
merge main and handle conflicts

shawnborton commented:
Here's another batch of illustrations to use: [MoreSimpleIllustrations.zip](https://github.com/Expensify/App/files/9975780/MoreSimpleIllustrations.zip)

I tried going through your screenshots in the same Figma file... here are some proposed ideas:

<img width="754" alt="image" src="https://user-images.githubusercontent.com/2319350/200960774-00e4846b-cf02-4da0-8808-cf933fdbf26f.png">

We might want to change the headline for this one, the equals sign made the break a bit awkward:
<img width="758" alt="image" src="https://user-images.githubusercontent.com/2319350/200960788-9adc68b7-6054-446a-a519-a96163bea86e.png">

<img width="743" alt="image" src="https://user-images.githubusercontent.com/2319350/200960835-48129162-6d16-4731-a641-f9a9b2b2ee05.png">

<img width="882" alt="image" src="https://user-images.githubusercontent.com/2319350/200960855-86aba140-c9d4-4064-878e-0bcf8ddd00f4.png">

<img width="917" alt="image" src="https://user-images.githubusercontent.com/2319350/200960874-4b46bc37-7876-48c0-8864-7893dd54ebe7.png">

<img width="855" alt="image" src="https://user-images.githubusercontent.com/2319350/200960884-e41cec9c-1c18-4d27-b1bb-d1e1fd3991b4.png">

<img width="883" alt="image" src="https://user-images.githubusercontent.com/2319350/200961132-03f84aa7-21d3-4a09-989f-7416f6bfae7b.png">


<img width="686" alt="image" src="https://user-images.githubusercontent.com/2319350/200961148-082ba7a9-3154-4bfe-a495-edeea1775176.png">


<img width="686" alt="image" src="https://user-images.githubusercontent.com/2319350/200961160-a2092540-bb2e-4a7a-8b54-152c1632ec2d.png">

dangrous commented:
Awesome, thanks @shawnborton ! I've updated all of the screens and so I'm going to go through and screenshot their final forms (and really try to get those last few). Now that we have updated ALL of them, I'm going to switch out the styling changes from each individual instance into the main `<Section>` component, too. Will post screenshots here, and then I think we just hold for the new colors/icons!

shawnborton commented:
Sounds great, thanks for the update! I'm planning to do a weekly update for the overall branding tasks in the open source room tomorrow, so let me know where we're at by tomorrow morning when you get a moment.

dangrous commented:
Haven't had time for anything yet but things are slowing down outside of this PR so I plan to have it completed and ready for final review later today!

shawnborton commented:
Sounds great, thanks for the update!

dangrous commented:
Ok have been thrown in a bunch of different directions today even though I was hoping not to be. Started going through this and will be able to finish tomorrow.

Commit:
add confirmed 50% of the new cards

Commit:
add all but 4 updated cards

dangrous commented:
Just committing some more updates here. Just the last four remaining (the harder ones to figure out where they actually exist... Can track progress here: https://docs.google.com/spreadsheets/d/1NEe1pc-1Wy2f0gpSCPPAb0SsHSAeXub56BuX_6-oaBU/edit#gid=0

Commit:
final components

dangrous commented:
Updated and ready for review! I believe this should remain on hold until the new colors/icons are ready because I'll likely have to make some final changes at that point. But, it's all synced up with the new style in all the sections. 

Screenshots, labelled by the number noted in the [Google Sheet](https://docs.google.com/spreadsheets/d/1NEe1pc-1Wy2f0gpSCPPAb0SsHSAeXub56BuX_6-oaBU/edit#gid=0) are [here](https://www.dropbox.com/s/1nv9cicwl758ea7/Screenshots.zip?dl=0) - all 138 of them. There are also a couple marked "OLD" which is for places I was able to get what they looked like before.

Lastly, I didn't make that copy change (the `=` one) because I figured we should wait for the new fonts in case the line break was different. I believe I made all the other changes, but take a look at the screenshots to make sure - you only really have to look at one of the platforms, they're pretty much all the same.

dangrous commented:
New reviewers - I think we can wait for review for now until @shawnborton and @grgia take a look but I wanted to take this out of draft to be safe.

dangrous commented:
Oops @shawnborton I found one more that definitely needs an update:

<img width="374" alt="image" src="https://user-images.githubusercontent.com/5487802/202303936-09ba22eb-0342-43d1-a89d-b908b0f4bfca.png">

Right now that's an animated icon (confetti pop) but obviously is the wrong size/background color, so something in the new style is probably preferable even without animation.

shawnborton commented:
How about this?
<img width="797" alt="image" src="https://user-images.githubusercontent.com/2319350/202324622-a828c2f9-a1cd-46b6-9582-7b83184c3aed.png">


No need to animate this, we can leave it as a static icon to match the others. 


[simple-illustration__thumbsupstars.svg.zip](https://github.com/Expensify/App/files/10026675/simple-illustration__thumbsupstars.svg.zip)

Commit:
add new illustration

Commit:
merge main

dangrous commented:
Updated to include that new image - thanks!

Commit:
remove old image import

shawnborton commented:
This is looking great, nice work Daniel!

The only small thing I noticed was on the pages below, it seems like the option rows are a bit too close together. I think they should all be 64px tall but have 0px spacing in between each of them. Let me know if that makes sense!

<img width="459" alt="image" src="https://user-images.githubusercontent.com/2319350/202599847-1e7a05c0-990c-4cee-9563-a698fa7be56d.png">

<img width="366" alt="image" src="https://user-images.githubusercontent.com/2319350/202599903-a9449c01-eef4-4806-bb36-6549849219b4.png">

<img width="366" alt="image" src="https://user-images.githubusercontent.com/2319350/202599990-b7a470ad-371f-4058-a12d-0dc7f39bc8bf.png">


It looks fine here though:
<img width="362" alt="image" src="https://user-images.githubusercontent.com/2319350/202599963-dc09ec61-3f3d-4042-a48e-88d88227a235.png">

grgia commented:
Okay, this looks SO GOOD

Commit:
update themecolor name and spacing

Commit:
remove testing code

dangrous commented:
I'm taking this off hold but only so that we can merge next week if we so desire, once we decide on a plan!

shawnborton commented:
Any reason why we couldn't merge this now, assuming everything looks good? cc @grgia @Luke9389 @JmillsExpensify

grgia commented:
IMO Since it's fairly contained, if it tests well on both current and dark theme, I don't see why not. If we decide to do that, I could merge the new IOU Preview card style too.

shawnborton commented:
That sounds great to me!

aldo-expensify commented:
We should merge `main` into this PR, this PR still contains the wrong colors that @grgia fixed

![image](https://user-images.githubusercontent.com/87341702/202806225-8dae73d5-fd8b-4557-8fd5-5b88b95f6053.png)

Reviewing like this may lead to confusions on if this is really breaking something or not.

aldo-expensify reviewed (commented):
Not sure if this is  intended or not:

<table>
<tr>
<td>
<img src="https://user-images.githubusercontent.com/87341702/202807683-092c7e31-394c-42fc-90ee-453506a94b8d.png">
</td>

<td>
<img src="https://user-images.githubusercontent.com/87341702/202807877-b9b43c6b-452a-433d-aa27-427864e95411.png">
</td>
</tr>
</table>

The first image shows the content wrapped in a different color "card", but in the second one there is not card wrapping. When do we wrap and when we don't?

Maybe maybe maybe it could be useful to accompany the testing/QA steps with screenshots to know what to expect. Or do you have a handy place to quickly see how things should look? Figma link?

dangrous commented:
That is as expected! That's the only place that doesn't have the card. Figma is here: https://www.figma.com/file/OgM6vzR98Qnr1gdmZOQPKz/Implement-Our-New-Brand?node-id=173%3A19885 and Screenshots are [dropbox.com/s/1nv9cicwl758ea7/Screenshots.zip?dl=0](https://www.dropbox.com/s/1nv9cicwl758ea7/Screenshots.zip?dl=0)

I will merge main and then we may need someone else to carry this over the line - about to run out to a flight!

Commit:
Merge branch 'main' into dangrous-workspace-editor

aldo-expensify commented:
Ha! you are right, it seems to be the only one not wrapped by a card. Is this intentional @shawnborton ?

@grgia do you want to take this over the finishing line in case there is anything else to do? otherwise I can give it a try.

shawnborton commented:
Yes, it is intentional!

Luke9389 commented:
Yeah I think we can definitely merge this now. It's completely independent of the other changes. 👍

JmillsExpensify commented:
+1. It's also a flow few users hit. Excited to get this on out!

aldo-expensify commented on src/pages/ReimbursementAccount/EnableStep.js:
```diff
@@ -126,16 +108,41 @@ class EnableStep extends React.Component {
+                                success
+                            />
+                        )}
+                        <View style={[styles.mv3]}>
+                            <MenuItem
+                                title={this.props.translate('workspace.bankAccount.disconnectBankAccount')}
+                                icon={Expensicons.Close}
+                                onPress={BankAccounts.requestResetFreePlanBankAccount}
+                                iconFill={themeColors.success}
+                                wrapperStyle={[styles.cardMenuItem]}
+                            />
+                        </View>
```
```suggestion
                        <MenuItem
                            title={this.props.translate('workspace.bankAccount.disconnectBankAccount')}
                            icon={Expensicons.Close}
                            onPress={BankAccounts.requestResetFreePlanBankAccount}
                            iconFill={themeColors.success}
                            wrapperStyle={[styles.cardMenuItem, styles.mv3]}
                        />
```

I think the wrapping `View` is not necessary, we can pass the style in `wrapperStyle` directly to the `MenuItem`

dangrous replied:
👍 I think your version is correct @aldo-expensify! This was left over from an earlier version where I was dealing with the heights differently. We should double check the presentation matches the Figma but this should be good.

aldo-expensify commented on src/pages/ReimbursementAccount/ValidationStep.js:
```diff
@@ -188,28 +189,32 @@ class ValidationStep extends React.Component {
+                                large
+                                success
+                            />
+                            <View style={[styles.mv3]}>
+                                <MenuItem
+                                    title={this.props.translate('workspace.bankAccount.noLetsStartOver')}
+                                    icon={Expensicons.RotateLeft}
+                                    onPress={BankAccounts.requestResetFreePlanBankAccount}
+                                    shouldShowRightIcon
+                                    iconFill={themeColors.success}
+                                    wrapperStyle={[styles.cardMenuItem]}
+                                />
+                            </View>
```
```suggestion
                            <MenuItem
                                title={this.props.translate('workspace.bankAccount.noLetsStartOver')}
                                icon={Expensicons.RotateLeft}
                                onPress={BankAccounts.requestResetFreePlanBankAccount}
                                shouldShowRightIcon
                                iconFill={themeColors.success}
                                wrapperStyle={[styles.cardMenuItem, styles.mv3]}
                            />
```
I think the wrapping `View` is not necessary, we can pass the style in `wrapperStyle` directly to the `MenuItem`

dangrous replied:
Same as above, this makes sense to me!

aldo-expensify commented on src/pages/workspace/card/WorkspaceCardVBANoECardView.js:
```diff
@@ -21,33 +23,38 @@ const WorkspaceCardVBANoECardView = props => (
                 {props.translate('workspace.card.checkingDomain')}
             </Text>
         )}
+
```
```suggestion
```

Unnecessary new line

aldo-expensify reviewed (commented):
I tested this in web, ios native and android native, and seems to be working fine overall. I left some minor comments.

In case it is useful for someone testing the PR: I tested some flows by hardcoding some values in variables. i.e. to get to the different steps of adding a bank account flow, or to see the different variations of WorkspaceCardPage:

<img width="524" alt="image" src="https://user-images.githubusercontent.com/87341702/202828300-f9a9be76-3a4e-42dc-b182-9b38e9667af3.png">
<img width="521" alt="image" src="https://user-images.githubusercontent.com/87341702/202828389-837be5e3-872c-4fcc-87db-09b18b0776dc.png">

dangrous committed:
Apply suggestions from code review

Co-authored-by: Aldo Canepa Garay <87341702+aldo-expensify@users.noreply.github.com>

dangrous commented:
@aldo-expensify I applied your suggestions, thank you! I can't build on this device (not work computer) so if you have time could you quickly check those two cards (Enable Step - both with and without the Expensify Card on the account - and Validation Step - "let's chat" stage) and make sure they still match the designs in the Figma? Really appreciate you taking the time to look through all these - I know it's a lot!

Santhosh-Sellavel commented:
That's a lot of screens it would take more time to test this one, I'll do my best to test this one with a couple of days!

aldo-expensify commented:
> @aldo-expensify I applied your suggestions, thank you! I can't build on this device (not work computer) so if you have time could you quickly check those two cards (Enable Step - both with and without the Expensify Card on the account - and Validation Step - "let's chat" stage) and make sure they still match the designs in the Figma? Really appreciate you taking the time to look through all these - I know it's a lot!

Thanks @dangrous , I'll be reviewing this today. Enjoy your time off!

aldo-expensify commented:
@JmillsExpensify I don't think this is WAQ, should we postpone spending time on reviewing this? The PR may be ready, but reviewing it thoroughly and providing screenshots of everything may be very time consuming considering the number of views affected.

shawnborton commented:
This is part of the branding initiative of the WAQ milestone.

aldo-expensify commented:
@shawnborton thanks, then I'll review today!

aldo-expensify commented:
## Reviewer Checklist

- [ ] I have verified the author checklist is complete (all boxes are checked off).
- [ ] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] iOS / native
    - [ ] Android / native
    - [ ] iOS / Safari
    - [ ] Android / Chrome
    - [ ] MacOS / Chrome
    - [ ] MacOS / Desktop
- [ ] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [ ] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [ ] I verified any copy / text that was added to the app is correct English and approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [ ] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn't already exist
    - [ ] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>

<!-- add screenshots or videos here -->

</details>

<details>
<summary>Mobile Web - Chrome</summary>

<!-- add screenshots or videos here -->

</details>

<details>
<summary>Mobile Web - Safari</summary>

<!-- add screenshots or videos here -->

</details>

<details>
<summary>Desktop</summary>

<!-- add screenshots or videos here -->

</details>

<details>
<summary>iOS</summary>

<!-- add screenshots or videos here -->

</details>

<details>
<summary>Android</summary>

<!-- add screenshots or videos here -->

</details>

aldo-expensify commented:
I finished taking screenshots for every platform, there are so many screenshots that I may have missed a few.

In my opinion, the PR looks look and doesn't appear to break anything. I would go ahead and merge this and if we see something that is not matching the designs, we fix it in a follow up PR.

Also, I think for a next time, it would be good to split the work done in smaller PR's (if possible).

aldo-expensify commented:
Thoughts @shawnborton @grgia ?

shawnborton commented:
I am down to merge and do follow ups for the small things we might have missed. Thanks so much for the review Aldo, I know this was a big one!

aldo-expensify commented:
@grgia do you want to review before I merge?

grgia reviewed (approved):
LGTM, tested dark mode

shawnborton commented:
@Santhosh-Sellavel all you!

Santhosh-Sellavel commented:
Thanks for the bump on it already!

Santhosh-Sellavel commented:
## Reviewer Checklist

- [ ] I have verified the author checklist is complete (all boxes are checked off).
- [ ] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] iOS / native
    - [ ] Android / native
    - [ ] iOS / Safari
    - [ ] Android / Chrome
    - [ ] MacOS / Chrome
    - [ ] MacOS / Desktop
- [ ] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [ ] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [ ] I verified any copy / text that was added to the app is correct English and approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [ ] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn't already exist
    - [ ] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>

https://user-images.githubusercontent.com/85645967/203452081-c8a45516-714c-4c83-8c86-05954ec7a220.mov

</details>

<details>
<summary>Mobile Web - Chrome</summary>

https://user-images.githubusercontent.com/85645967/203452607-62f4fe77-ce2c-4772-9608-ca47aa76a404.mp4

</details>

<details>

<summary>Mobile Web - Safari</summary>

https://user-images.githubusercontent.com/85645967/203451990-432d2b20-7d81-42ab-b2d6-f702d81a81d4.MP4

</details>

<details>
<summary>Desktop</summary>

https://user-images.githubusercontent.com/85645967/203451967-53036796-d418-46f8-9e29-9e5f8e59adbe.mov

</details>

<details>
<summary>iOS & Android</summary>

https://user-images.githubusercontent.com/85645967/203451931-69ded98a-9826-4ff9-95b0-655e78e757e9.mov

</details>

Santhosh-Sellavel commented:
Is this expected on offline, shouldn't buttons be disabled?
 cc: @shawnborton 
![Screenshot_1669166715](https://user-images.githubusercontent.com/85645967/203452774-ce4f589a-15ed-4e0d-a04d-861d576cc16c.png)

shawnborton commented:
Hmm good catch, how does it currently work on the main branch?

Santhosh-Sellavel commented:
Same behaviour on main too!

shawnborton commented:
Okay cool, then no worries if it's the same thing on main.

Santhosh-Sellavel commented:
Should we log a bug for it @shawnborton ?

Santhosh-Sellavel reviewed (approved):
Looks good!

shawnborton commented:
@Santhosh-Sellavel can't hurt, you can bring that up in #expensify-bugs

grgia commented:
If the bug is logged, am I good to merge this?

aldo-expensify commented:
Lol, now I see that Luke posted "Buttons merged", so I guess that was about a different PR

shawnborton commented:
Yup, let’s do it!

On Wed, Nov 23, 2022 at 10:47 AM Aldo Canepa Garay ***@***.***>
wrote:

> @shawnborton <https://github.com/shawnborton> from this slack
> <https://expensify.slack.com/archives/C01GTK53T8Q/p1669147855166289>
> conversation , I understand that we should proceed with the merge, right?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/Expensify/App/pull/12419#issuecomment-1325515645>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AARWH5R3RLD2IVAOVC3UDLLWJZRCHANCNFSM6AAAAAARVUSZKQ>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

aldo-expensify commented:
Done!

OSBotify commented:
:hand: This PR was not deployed to staging yet because QA is ongoing. It will be automatically deployed to staging after the next production release.

OSBotify commented:
🚀 [Deployed](https://github.com/Expensify/App/actions/runs/3536513612) to staging by @aldo-expensify in version: 1.2.31-0 🚀

 platform | result 
 ---|--- 
🤖 android 🤖|success ✅ 
🖥 desktop 🖥|success ✅
🍎 iOS 🍎|success ✅ 
🕸 web 🕸|success ✅

luacmartins commented:
The production deploy comment failed for this PR, but this was deployed to production on v1.2.32-2 on Nov 28.

dangrous commented:
As requested in https://github.com/Expensify/App/issues/13318#issuecomment-1349860737 - commenting here. This was not really a bug per se so we don't need a new process. Generally I guess we should make sure that if multiple PRs are affecting something (say, a redesign), we make sure that all of them are considered in each. In this case, we were matching the style of the old buttons, and not the new ones.
